### ISSUE TITLE
[Enhancement] Documentation for Solution/Project structure for contributors 

### ISSUE THREAD

**Issue by EmilAlipiev** [xamarin/Xamarin.Forms#11307]
I try to contribute and i want to contribute more but i spend a lot of time to do reverse engineering trying to understand whole this solution/project structure. I would be so nice to have a short wiki docu which explains each and every project, what they are used for, how they are referenced.
Maybe some key renderers how they are created. How does platformspecific features work. 
I mean just drop some notes. what is Xamarin.Core librar for. What is ControlGallery for, how should we use it?
When we fix an issue, how should we create a unit test for it etc. etc. 
Project has not explanatory comments also. I would put some function comments, some class comments for people to understand it better.

**Comment by pictos** [comment]
@EmilAlipiev  you can find more info here:
https://devblogs.microsoft.com/xamarin/beginners-guide-contributing-xamarin-forms/
https://javiersuarezruiz.wordpress.com/2019/08/07/como-colaborar-con-xamarin-forms/

I agree that we can have a better doc in the repo.

**Comment by EmilAlipiev** [comment]
@pictos that looks cool. I didnt see this before from javier. it looks more up to date.

**Comment by jfversluis** [comment]
In glad this got you unblocked @EmilAlipiev but I'd be happy to discuss this a little further if you don't mind. It would be awesome if we could have complete docs that are easily discoverable right off the bat when you want to start contributing.

Maybe we can stay up a thread under the discussions where I'd like to hear from you what places you looked for the docs and what information would help you get underway. Would you be willing to discuss that a bit more?

**Comment by EmilAlipiev** [comment]
Hi @jfversluis 

I think that should be on the official Microsoft docs where we check documentation for all about xamarin.forms. 
For me there are 2 big challenges on contributing xamarin.forms. 
1) Fork vs Clone; It is tedious to work with a fork if you want to send PR. First i started with fork and you need to create branch on your fork and probably merge in your master fork to test if you have multiple fork branches to see if they are working all together. Then you should PR for each. I understand that you cannot allow everyone to work on a clone and PR directly. Like james mention's here https://youtu.be/TSi9cFRFuN4?t=398 this confusion occurs every time for us. Video is good indeed, i wish it was done long time ago. 
2) Source code has no single comment to explain what is what. For example, i have been reverse engineering CollectionView recently and it is complicated structure from CollectionViewRenderer->GroupableItemsViewRenderer->SelectableItemsViewRenderer->StructuredItemsViewRenderer->ItemsViewRenderer->ViewRenderer->VisualElementRenderer and finally inherits native Panel element of UWP.
I wish there was comments explaining each those renderers. People would waste less time and you would know that on which renderers OnElementPropertyChanged you should implement when Item is changed or entire layout is changed etc.

**Comment by jfversluis** [comment]
I've moved your message to the discussion [I opened for this](https://github.com/xamarin/Xamarin.Forms/discussions/11320). If you'd like to discuss a bit more there and maybe look at the ideas I jotted down there that would be great :)
